File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.java
import java.util.Scanner;

class Main{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int target = sc.nextInt();
        int startP = 1;
        int endP = 1;
        int sum = 1;
        int count = 1;
        while(endP<target) {
            if(sum < target){
                endP++;
                sum += endP;
            } else if(sum > target) {
                sum -= startP;
                startP++;
            } else if(sum == target) {
                endP++;
                sum += endP;
                count++;
            }
        }
        System.out.println(count);
    }
}

File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.go
package main

import "fmt"

func main() {
	var t int
	fmt.Scan(&t)
	sp := 1
	ep := 1
	sum := 1
	count := 1
	for ep < t {
		if sum < t {
			ep++
			sum += ep
		} else if sum > t {
			sum -= sp
			sp++
		} else if sum == t {
			ep++
			sum += ep
			count++
		}
	}
	fmt.Println(count)
}
